doc/user/sql/create-source/avro-source.md

```diff
@@ -0,0 +1,188 @@
+Materialize.
+
+### Kafka source details
+
+Materialize expects each source to use to one Kafka topic, which is&mdash;in
+  turn&mdash;generated by a single table in an upstream database.
+
+### Avro format details
+
+Avro-formatted external sources require providing you providing the schema in
+one of two ways:
+
+- Using the [Confluent Schema Registry](#using-a-confluent-schema-registry)
+- Providing the Avro schema [in-line when creating the
+  source](#inlining-the-avro-schema).
+
+### Envelope details
+
+Envelopes determine whether an incoming record inserts new data, updates or
+deletes existing data, or both. For more information, see [API Components:
+Envelopes](../../../overview/api-components#envelopes).
+
+Avro-encoded data sent through Kafka is the only type of data that can support
+the Debezium envelope.
+
+### Debezium envelope details
+
+The Debezium envelope provides a "diff envelope", which describes the decoded
+records' old and new values; this is roughly equivalent to the notion of Change
+Data Capture, or CDC. Materialize can use the data in this diff envelope to
+process data as representing inserts, updates, or deletes.
+
+This envelope is called the Debezium envelope because the only tool that we're
+aware of that includes this data is [Debezium].
+
+To use the Debezium envelope with Materialize, you must configure Debezium with
+your database.
+
+- [MySQL](https://debezium.io/documentation/reference/0.10/connectors/mysql.html)
+- [PostgreSQL](https://debezium.io/documentation/reference/0.10/connectors/postgresql.html)
+
+The Debezium envelope is only supported by sources published to Kafka by
+Debezium.
+
+#### Format implications
+
+Using the Debezium envelopes changes the schema of your Avro-encoded Kafka
+topics to include something akin to the following field:
+
+```json
+{
+    "type": "record",
+    "name": "envelope",
+    "fields": [
+        {
+        "name": "before",
+        "type": [
+            {
+            "name": "row",
+            "type": "record",
+            "fields": [
+                {"name": "a", "type": "long"},
+                {"name": "b", "type": "long"}
+            ]
+            },
+            "null"
+        ]
+        },
+        { "name": "after", "type": ["row", "null"] }
+    ]
+}
+```
+
+Note that:
+
+- If you use the Confluent Schema Registry to receive your schemas, you don't
+  need to manually create this field; Debezium will have taken care of it for
+  you.
+- The following section depends on the column's names and types, and is unlikely
+  to match our example:
+    ```json
+    ...
+    "fields": [
+            {"name": "a", "type": "long"},
+            {"name": "b", "type": "long"}
+        ]
+    ...
+    ```
+
+## Examples
+
+### Using a Confluent schema registry
+
+```sql
+CREATE SOURCE events
+FROM KAFKA BROKER 'localhost:9092' TOPIC 'events'
+FORMAT AVRO
+    USING CONFLUENT SCHEMA REGISTRY 'http://localhost:8081';
+```
+
+This creates a source that...
```

Comment:
I would have used `:` instead of `...`, but you definitely know more than me about copy-editing conventions, so take it or leave it.